pkg/gameplay/game.go

```diff
@@ -401,11 +410,25 @@ func HandleEvent(ctx context.Context, gameStore GameStore, userStore user.Store,
 	timeRemaining := entGame.TimeRemaining(onTurn)
 	log.Debug().Int("time-remaining", timeRemaining).Msg("checking-time-remaining")
 	// Check that we didn't run out of time.
-	if entGame.TimeRanOut(onTurn) {
+	// Allow auto-passing.
+	if !(entGame.Game.Playing() == macondopb.PlayState_WAITING_FOR_FINAL_PASS &&
+		cge.Type == pb.ClientGameplayEvent_PASS) &&
+		entGame.TimeRanOut(onTurn) {
 		// Game is over!
 		log.Debug().Msg("got-move-too-late")
-		// Basically skip to the bottom and exit.
-		return entGame, setTimedOut(ctx, entGame, onTurn, gameStore, userStore, listStatStore)
+
+		// If an ending game gets "challenge" just before "timed out",
```

Comment:
This part of the code gets triggered when an event happens after time has run out. So if the event comes in too late, and it's a challenge, just make it a pass (rather than a resign).

Reply:
Ah, okay, sounds good.